horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/AlertingServiceImpl.java

```diff
@@ -494,13 +494,11 @@ private void emitDatapoints(DatasetDAO dataset, boolean notify, boolean debug, R
       );
       if (!missingValueVariables.isEmpty()) {
          MissingValuesEvent event = new MissingValuesEvent(dataset.getInfo(), missingValueVariables, notify);
-         if(mediator.testMode())
-            Util.registerTxSynchronization(tm, txStatus -> messageBus.publish(MessageBusChannels.DATASET_MISSING_VALUES, dataset.testid, event));
+         mediator.publishEvent(AsyncEventChannels.DATASET_MISSING_VALUES, dataset.testid, event);
          mediator.missingValuesDataset(event);
       }
       DataPoint.DatasetProcessedEvent event = new DataPoint.DatasetProcessedEvent( DatasetMapper.fromInfo( dataset.getInfo()), notify);
-      if(mediator.testMode())
```

Comment:
i don't understand, the test has been moved to inside the method: https://github.com/Hyperfoil/Horreum/pull/1493/files#diff-fef1cb67fede2ae3db1c97957a6f1e71b2be10ae68b546fa28aad09c220e0f1eR234

It is the same behaviour, but without the call to `mediator.testMode()` repeated throughout the codebase.

At runtime, it will always be a no-op and the JIT will inline the call to a no-op.

There are no new object created, but if we want to conserve memory, may the allocation `DataPoint.Event event = new DataPoint.Event(DataPointMapper.from( dataPoint), dataset.testid, notify);`  should happen within a `testMode()` check, rather than being allocated each time?

Reply:
Yes, but you will still make the call to Util.registerTxSynchronization as well right?

Reply:
@stalep I have reinstated all the `if(mediator.testMode())` checks